qiskit/transpiler/passes/consolidate_blocks.py

```diff
@@ -62,7 +62,16 @@ def run(self, dag):
             if not isinstance(wire, Qubit):
                 continue
             global_qregs = list(dag.qregs.values())
-            global_index_map[wire] = global_qregs.index(wire.register) + wire.index
+
+            reg_index = 0
+            for g_qreg in global_qregs:
+                # if this is the register we want, stop
+                if g_qreg == wire.register:
+                    break
+                # index needs to increase by the size of the register
+                reg_index += g_qreg.size
+
+            global_index_map[wire] = reg_index + wire.index
```

Comment:
To make sure I understand, it looks like `global_index_map` should be a mapping from every `Qubit` wire in `dag.wires` to an integer, its index in the ordered list of (register, wire) pairs, set by the register ordering in `dag.qregs.values()` and the index ordering of the wire within its register. Is that right?

If so, could this be simplified by laying out all the `Qubit` wires in the desired order first and building the mapping from the indices of that list? Maybe something like `global_index_map = { wire, idx for idx, wire in enumerate(dag.qubits()) }` would be equivalent?

Reply:
Yes, you are correct. I see that `dag.qubits()` uses the `qregs.values()` ordering so this should be equivalent.

Reply:
Updated in cbfd528